ui/apps/platform/src/Containers/Vulnerabilities/PlatformCves/PlatformCve/PlatformCvePage.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from 'react';
+import { PageSection, Breadcrumb, Divider, BreadcrumbItem, Skeleton } from '@patternfly/react-core';
+import { useParams } from 'react-router-dom';
+
+import PageTitle from 'Components/PageTitle';
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+
+import { getOverviewPagePath } from '../../utils/searchUtils';
+import CvePageHeader, { CveMetadata } from '../../components/CvePageHeader';
+
+const workloadCveOverviewCvePath = getOverviewPagePath('Platform', {
+    entityTab: 'CVE',
+});
+
+function PlatformCvePage() {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { cveId } = useParams() as { cveId: string };
```

Comment:
I'd say in this case, the first is safer, as much as my general opinion is "never use the `as` keyword". Given the way routes are handled, this page will never be rendered if `:cveId` in the route URL is not matched so the `undefined` case can never occur. We then get the correct runtime type (`string`) of `:cveId` further down in the code. React router does not parse values into numbers, booleans, etc. by default, as far as I can tell, so regardless of what `:cveId` is it will always be a string here. It looks like we will get full type-safety, but almost coincidentally.

The second case would actually give us less type safety since we do not have types for `react-router` installed. This will silently consider `cveId` as type `any` without any warning. Although even if we did have types installed, I imagine React router handles generics similarly to axios, which is essentially doing an `as` cast under the hood. If the response from the server (or the URL parameter) is correct, we get more coincidental type safety elsewhere in the code. If the data is _not_ correct, we don't get any error at compile time and the value is assumed to be whatever we declared in the generic, regardless of what it actually is.

The type-safest way here is probably something like:
```tsx
const { cveId } = useParams();
if (typeof cveId !== 'string') {
    return <div>Oh no!</div>;
}
// cveId is correctly inferred to be `string` from here on out
```

I've heard newer frameworks like the Tanstack router have fully type-safe routing built in, but I haven't looked into it myself.